prow/jenkins/jenkins_test.go

```diff
@@ -177,3 +178,85 @@ func TestListBuilds(t *testing.T) {
 		}
 	}
 }
+
+func TestBuildCreate(t *testing.T) {
+	jobSpecs := []prowapi.ProwJobSpec{
+		{
+			Agent: "jenkins",
+			Job:   "my-jenkins-job-name",
+			JenkinsJob: &prowapi.JenkinsJob{
+				GithubBasedProject: true,
+			},
+			Refs: &prowapi.Refs{
+				BaseRef: "master",
+				BaseSHA: "deadbeef",
+				Pulls: []prowapi.Pull{
+					{
+						Number: 123,
+						SHA:    "abcd1234",
+					},
+				},
+			},
+		},
+		{
+			Agent: "jenkins",
+			Type:  prowapi.PostsubmitJob,
+			Job:   "my-jenkins-job-name",
+			JenkinsJob: &prowapi.JenkinsJob{
+				GithubBasedProject: true,
+			},
+			Refs: &prowapi.Refs{
+				BaseRef: "master",
+				BaseSHA: "deadbeef",
+			},
+		},
+		{
+			Agent: "jenkins",
+			Job:   "my-k8s-job-name",
+			Refs: &prowapi.Refs{
+				BaseRef: "master",
+				BaseSHA: "deadbeef",
+				Pulls: []prowapi.Pull{
+					{
+						Number: 123,
+						SHA:    "abcd1234",
+					},
+				},
+			},
+		},
+	}
+
+	expectedPaths := []string{
+		"/job/my-jenkins-job-name/view/change-requests/job/PR-123/buildWithParameters",
+		"/job/my-jenkins-job-name/job/master/buildWithParameters",
+		"/job/my-k8s-job-name/buildWithParameters",
+	}
+
+	for idx, jobSpec := range jobSpecs {
+		var path string
+
+		var handler http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
+			fmt.Println(r.Method)
```

Comment:
You could use `t` to check these values (and the path) during the call if you want, too

Reply:
Oh, these `fmt`s are leftover, don't need them here.